File: 1464-maximum-product-of-two-elements-in-an-array/1464-maximum-product-of-two-elements-in-an-array.py
class Solution:
    def maxProduct(self, nums: List[int]) -> int:
        max1 = None
        max2 = None
        for n in nums:
            if max1 is None:
                max1 = n
            elif max2 is None:
                if n > max1:
                    max1, max2 = n, max1
                else:
                    max2 = n
            elif n > max1:
                max1, max2 = n, max1
            elif n > max2:
                max2 = n
        return (max1 - 1) * (max2 - 1)

File: 1464-maximum-product-of-two-elements-in-an-array/1464-maximum-product-of-two-elements-in-an-array.cpp
class Solution {
public:
    int maxProduct(vector<int>& nums) {
        int max1 = 0;
        int max2 = 0;
        int n = 0;
        for (int i=0; i<nums.size(); i++) {
            n = nums[i];
            if (not(max1)) {
                max1 = n;
            }
            else if (not(max2)) {
                if (n > max1) {
                    max2 = max1;
                    max1 = n;
                }
                else {
                    max2 = n;
                }
            }
            else if (n > max1) {
                max2 = max1;
                max1 = n;
            }
            else if (n > max2) {
                max2 = n;
            }
        }
        cout << max1;
        cout << max2;
        return (max1 - 1) * (max2 - 1);
    }
};